### DISCUSSION TITLE
clairctl binary for alpine images?

### DISCUSSION THREAD

**Discussion by mieliespoor** [quay/clair#1465] [General]
clairctl is currently available in binary form for linux-386, darwin-amd64 and linux-amd64. Only clairctl-linux-386 however runs in a linux alpine container image. Would it be possible to build a binary that would work on alpine so it won't be needed to fall back on the clairctl-linux-386 binary?

As an example, when trying to run the clairctl-linux-amd64 binary in the alpine image, I get the following (It is the same for the darwin binary):

```
$ docker run -it consoleapp2:latest /bin/sh
/app # ls -al
total 22544
drwxr-xr-x    1 root     root          4096 Dec 31 19:34 .
drwxr-xr-x    1 root     root          4096 Dec 31 19:34 ..
-rwxr-xr-x    1 root     root         89672 Dec 31 19:26 ConsoleApp2
-rw-r--r--    1 root     root           403 Dec 31 19:26 ConsoleApp2.deps.json
-rw-r--r--    1 root     root          6144 Dec 31 19:26 ConsoleApp2.dll
-rw-r--r--    1 root     root          9656 Dec 31 19:26 ConsoleApp2.pdb
-rw-r--r--    1 root     root           146 Dec 31 19:26 ConsoleApp2.runtimeconfig.json
-rwxr-xr-x    1 root     root      22946889 Dec 30 15:16 clairctl-linux-amd64
-rwxr-xr-x    1 root     root           188 Dec 31 19:33 clairctl.yml
/app # ./clairctl-linux-amd64 -v
/bin/sh: ./clairctl-linux-amd64: not found
```
as mentioned, the 386 binary works
```
/app # ./clairctl-linux-386
NAME:
   clairctl - interact with a clair API

USAGE:
   clairctl-linux-386 [global options] command [command options] [arguments...]

VERSION:
   0.2.0
```

**Comment by hdonnay** [comment]
I think you need the glibc compatibility layer for alpine. Alternatively, we could build a no-cgo version of the tool, but I'm hesitant to do that as it would no longer work the same way as a "normal" program on a normal distro.
